customers/views.py

```diff
@@ -34,24 +36,28 @@ def create():
             "email": f"{email}",
             "dob": f"{dob}",
             "date_created": f"{newCustomer.date_created}",
-        }
+        }, 200
     except exc.IntegrityError:
         # Error Code 409, for resource already exists / duplicate resource / Conflict
         # Error that occurs when a customer already has a place in the database
-        return {"Message": "ERROR, Code: 409, Customer already exists"}
+        # Should occur when a customer enters a unique field such as email and username more than once
+        return {"Message": "ERROR, Code: 409, Customer already exists"}, 409
     except KeyError:
         # Error that occurs when a user leaves fields empty
-        return {"Message": "ERROR, Code: 422, left required fields empty"}
+        return {"Message": "ERROR, Code: 422, left required fields empty"}, 422
 
 
-def retrieve(customer_id=-1):
+@views.route("/list", defaults={"customer_id": None}, methods=["GET"])
+@views.route("/detail/", defaults={"customer_id": None}, methods=["GET"])
+@views.route("/detail/<int:customer_id>", methods=["GET"])
+def retrieve(customer_id):
     try:
-        if customer_id == -1:
+        if customer_id is None:
             customer_list = {}
             for i in range(1, Customer.query.count() + 1):
                 selected_customer = Customer.query.filter_by(id=i).first()
                 customer_list.update({f"Customer {i}": f"{selected_customer.username}"})
```

Comment:
this should be returning the same object as `create` and `detail` would. Also, isn't there a query that returns all customers?